docs/tutorials/10_effective_dimension.ipynb

```diff
@@ -0,0 +1,763 @@
+    "    else:\n",
+    "        plt.plot(x[0], x[1], \"go\")\n",
+    "plt.plot([-1, 1], [1, -1], \"--\", color=\"black\")\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The next step is to create a QNN, an instance of `TwoLayerQNN` in our case. Since we pass only the number of inputs, the network will continue with the default values for feature map and ansatz."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "opflow_qnn = TwoLayerQNN(num_inputs, quantum_instance=qi_sv)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The training step may take some time, be patient. You can pass a callback to the classifier to observe how the training process is going on. We fix `initial_point` for reproducibility purposes as usual."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "0.75"
+      ]
+     },
+     "execution_count": 16,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "initial_point = algorithm_globals.random.random(opflow_qnn.num_weights)\n",
+    "\n",
+    "# construct classifier\n",
+    "circuit_classifier = NeuralNetworkClassifier(\n",
+    "    neural_network=opflow_qnn, optimizer=COBYLA(), initial_point=initial_point\n",
+    ")\n",
+    "\n",
+    "# fit classifier to data\n",
+    "circuit_classifier.fit(X, y)\n",
+    "circuit_classifier.score(X, y)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, we evaluate local effective dimension using only one randomly sampled set of weights."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "random_weights = algorithm_globals.random.uniform(0, 1, size=(1, opflow_qnn.num_weights))\n",
+    "\n",
+    "local_ed_random = LocalEffectiveDimension(\n",
+    "    qnn=opflow_qnn, weight_samples=random_weights, input_samples=X  # send only one set of weights\n",
+    ")\n",
+    "local_eff_dim_random = local_ed_random.get_effective_dimension(dataset_size=n)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In contrast to what we did in the previous step, here we evaluate effective dimension based on the trained weights. To do that we access the trained weights directly from the classifier."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "trained_weights = circuit_classifier._fit_result.x\n",
```

Comment:
I guess this is part of the rationale for #385 to avoid accessing private field

Reply:
> I guess this is part of the rationale for #385 to avoid accessing private field

This is correct.